src/stp/plays/referee_specific/PenaltyUs.cpp

```diff
@@ -4,69 +4,84 @@
 
 #include "include/roboteam_ai/stp/plays/referee_specific/PenaltyUs.h"
 
-#include "stp/invariants/game_states/PenaltyUsGameStateEvaluation.h"
-#include "include/roboteam_ai/stp/roles/active/Attacker.h"
-#include "include/roboteam_ai/stp/roles/passive/Halt.h"
-#include "stp/roles/PenaltyKeeper.h"
+#include <stp/roles/Keeper.h>
+#include <stp/roles/PenaltyKeeper.h>
+#include <stp/roles/active/Attacker.h>
+#include <stp/roles/passive/Halt.h>
+
+#include "include/roboteam_ai/stp/computations/PositionComputations.h"
 
 namespace rtt::ai::stp::play {
+const char* PenaltyUs::getName() { return "Penalty Us"; }
 
 PenaltyUs::PenaltyUs() : Play() {
-    startPlayInvariants.clear();
-    startPlayInvariants.emplace_back(std::make_unique<invariant::PenaltyUsGameStateInvariant>());
-
-    keepPlayInvariants.clear();
-    keepPlayInvariants.emplace_back(std::make_unique<invariant::PenaltyUsGameStateInvariant>());
-
-    roles = std::array<std::unique_ptr<Role>, stp::control_constants::MAX_ROBOT_COUNT>{std::make_unique<role::PenaltyKeeper>(role::PenaltyKeeper("keeper")),
-                                                                                       std::make_unique<role::Attacker>(role::Attacker("kicker")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_0")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_1")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_2")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_3")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_4")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_5")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_6")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_7")),
-                                                                                       std::make_unique<role::Halt>(role::Halt("halt_8"))};
+    /// Evaluations that have to be true to be considered when changing plays.
+    startPlayEvaluation.clear(); // DONT TOUCH.
+    startPlayEvaluation.emplace_back(eval::NormalPlayGameState);
+
+    /// Evaluations that have to be true to allow the play to continue, otherwise the play will change. Plays can also end using the shouldEndPlay().
+    keepPlayEvaluation.clear(); // DONT TOUCH.
+    keepPlayEvaluation.emplace_back(eval::NormalPlayGameState);
```

Comment:
The same thing: `keepPlayEvaluation.emplace_back(eval::PenaltyUsGameState);`

Reply:
fixed